ipn/ipnlocal/local.go

```diff
@@ -342,7 +342,17 @@ func (b *LocalBackend) onHealthChange(sys health.Subsystem, err error) {
 // can no longer be used after Shutdown returns.
 func (b *LocalBackend) Shutdown() {
 	b.mu.Lock()
+	if b.shutdownCalled {
+		b.mu.Unlock()
+		return
+	}
 	b.shutdownCalled = true
+
+	if b.loginFlags&controlclient.LoginEphemeral != 0 {
+		b.mu.Unlock()
+		b.logout(b.ctx, true) // best effort
```

Comment:
can we bound the time here? Or start it async in its own goroutine?

Not a fan of blocking Shutdown with network work.

Reply:
Time bound sgtm
Running a different goroutine doesn't help shutdown